bobcat/bb-cumber/src/main/java/com/cognifide/qa/bb/cumber/Bobcumber.java

```diff
@@ -161,9 +129,39 @@ private void closeWebDriverPool() {
     }
   }
 
-  private boolean isFeatureFileEmpty() throws IOException {
-    try (BufferedReader br = new BufferedReader(new FileReader(file))) {
-      return br.readLine() == null;
+  private void shutdownRerun(RunNotifier notifier) {
+    try {
+      double percentageOfFailedTests = statisticsHelper.getPercentageOfFailedTests(statisticsFile);
+      int failedTestsNumber = statisticsHelper.getNumberOfFailedTests(statisticsFile);
+      if (failedTestsNumber == 0) {
+        notifier.fireTestFinished(Description.EMPTY);
+      } else if (percentageOfFailedTests > maxFailedTestPercentage) {
+        String failureMessage = "Percentage of failed tests was bigger than " + maxFailedTestPercentage + ".";
+        Failure failure = new Failure(Description.createSuiteDescription(failureMessage), new TooManyTestsToRerunException(failureMessage));
+        notifier.fireTestFailure(failure);
+      }
+    } catch (FileNotFoundException e) {
+      LOG.error("Statistics file not found.", e);
+    }
+  }
+
+  private boolean canRerunFailedTests() {
+    boolean statement = true;
+    try {
+      int failedTestsNumber = statisticsHelper.getNumberOfFailedTests(statisticsFile);
+      double percentageOfFailedTests = statisticsHelper.getPercentageOfFailedTests(statisticsFile);
+      statement = !((failedTestsNumber == 0) || (percentageOfFailedTests > maxFailedTestPercentage));
+    } catch (FileNotFoundException e) {
```

Comment:
you are catching the same exception two times - please consider file problems handling as a responsibility of statistics helper that could return some neutral value and log error or something like that